crates/lib/src/execution_data.rs

```diff
@@ -232,27 +243,27 @@ impl ReadoutMap {
                         BTreeMap<MemoryReference, &ReadoutValues>,
                         RegisterMatrixConversionError,
                     >>()?
-                    .iter()
+                    .into_iter()
                     // Iterate over them in reverse. Starting with the last index lets us:
                     //     (1): Initialize the RegisterMatrix with the correct number of rows
                     //     (2): Use a memory reference with an index of 0 as a setinel.
                     .rev()
                     .try_fold(
                         (
-                            HashMap::with_capacity(readout_data.readout_values.len()),
-                            None::<&MemoryReference>,
+                            HashMap::with_capacity(qpu_result_data.readout_values.len()),
+                            None::<MemoryReference>,
                         ),
                         |(mut readout_data, previous_reference), (reference, values)| {
                             // If we haven't started on a new register, make sure we aren't
                             // skipping an index. For example, if we jumped from ro[5] to ro[3], or
-                            // ro[2] to theta[1], we skipped a column.
+                            // ro[2] to theta[0], we skipped a column.
                             if let Some(previous) = previous_reference {
                                 if previous.name != reference.name
-                                    || previous.index != reference.index + 1
+                                    || previous.index - 1 != reference.index
```

Comment:
Still unsure about the logic here. Shouldn't it be this?
```suggestion
                                    || previous.index + 1 != reference.index
```

i.e., we expect `previous.index == reference.index - 1` or `previous.index + 1 == reference.index`?

Reply:
We are iterating over the references in reverse, so if we evaluate some index `n`, then the correct index to see next would be `n - 1`.  Hence, the check that the current `reference.index` is equal to `previous.index - 1`. If not, then we return an error with `previous.index - 1`, since that is the index we expected to see.